modules/le_log/le_log.cpp

```diff
@@ -11,23 +11,28 @@ struct le_log_module_o {
 	std::atomic_int log_level = 1;
 };
 
+struct le_log_context_o {
+	le_log_module_o                                    module_default;
+	std::unordered_map<std::string, le_log_module_o *> modules;
+	std::mutex                                         mtx;
+};
+
+static le_log_context_o *ctx;
+
 static le_log_module_o *le_log_module_default() {
-	static auto module = new le_log_module_o();
-	return module;
+	return &ctx->module_default;
 }
 
 static le_log_module_o *le_log_get_module( const char *name ) {
```

Comment:
What happens if this gets called with `le_log_get_module("")` (empty string)? I think it could be good to then return the default le_log module

Reply:
I suggest that we crash the whole system and display a message "trying to outsmart me, arr" :)
no, makes sense. I have added a check